Running of tests is not independent of current working directory

Hi,

I try to test this plugin with the shipped tests, but failed, see following for more info.

```
(aiida-py38) werner@ubuntu-01:~/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests$ pytest
============================= test session starts ==============================
platform linux -- Python 3.8.2, pytest-5.4.1, py-1.5.4, pluggy-0.13.1
rootdir: /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests, inifile: pytest.ini
plugins: cov-2.8.1, timeout-1.3.4
collected 87 items                                                             

calculations/test_autoinvalidate_cache.py ..............                 [ 16%]
calculations/test_epw.py E                                               [ 17%]
calculations/test_helpers.py .                                           [ 18%]
calculations/test_matdyn.py E                                            [ 19%]
calculations/test_ph.py EE                                               [ 21%]
calculations/test_pw.py E                                                [ 22%]
calculations/test_pw2gw.py E                                             [ 24%]
parsers/test_cp.py E                                                     [ 25%]
parsers/test_dos.py E.                                                   [ 27%]
parsers/test_matdyn.py E                                                 [ 28%]
parsers/test_neb.py EE.                                                  [ 32%]
parsers/test_ph.py EEE                                                   [ 35%]
parsers/test_projwfc.py E                                                [ 36%]
parsers/test_pw.py EEEEE...EEE.EEEE..EE..........                        [ 71%]
parsers/test_pw2gw.py E...                                               [ 75%]
parsers/test_pw2wannier90.py E                                           [ 77%]
parsers/test_q2r.py E                                                    [ 78%]
tools/test_immigrate.py F                                                [ 79%]
tools/calculations/test_pw.py .                                          [ 80%]
utils/test_bands.py ....                                                 [ 85%]
utils/test_convert.py ........                                           [ 94%]
utils/test_restart.py .....                                              [100%]

==================================== ERRORS ====================================
______________________ ERROR at setup of test_epw_default ______________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_epw.py, line 13
  def test_epw_default(
E       fixture 'file_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_epw.py:13
____________________ ERROR at setup of test_matdyn_default _____________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_matdyn.py, line 16
  def test_matdyn_default(
E       fixture 'file_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_matdyn.py:16
______________________ ERROR at setup of test_ph_default _______________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_ph.py, line 15
  def test_ph_default(
E       fixture 'file_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_ph.py:15
____________________ ERROR at setup of test_ph_qpoint_list _____________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_ph.py, line 55
  def test_ph_qpoint_list(
E       fixture 'file_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_ph.py:55
______________________ ERROR at setup of test_pw_default _______________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_pw.py, line 11
  def test_pw_default(
E       fixture 'file_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_pw.py:11
______________________ ERROR at setup of test_pw_default _______________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_pw2gw.py, line 11
  def test_pw_default(
E       fixture 'file_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/calculations/test_pw2gw.py:11
______________________ ERROR at setup of test_cp_default _______________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_cp.py, line 21
  def test_cp_default(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_cp.py:21
______________________ ERROR at setup of test_dos_default ______________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_dos.py, line 14
  def test_dos_default(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_dos.py:14
____________________ ERROR at setup of test_matdyn_default _____________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_matdyn.py, line 19
  def test_matdyn_default(aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, data_regression):
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_matdyn.py:19
______________________ ERROR at setup of test_neb_default ______________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_neb.py, line 50
  def test_neb_default(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_neb.py:50
__________________ ERROR at setup of test_neb_all_iterations ___________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_neb.py, line 82
  def test_neb_all_iterations(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_neb.py:82
______________________ ERROR at setup of test_ph_default _______________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_ph.py, line 13
  def test_ph_default(aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, data_regression):
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_ph.py:13
___________________ ERROR at setup of test_ph_not_converged ____________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_ph.py, line 30
  def test_ph_not_converged(aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, data_regression):
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_ph.py:30
__________________ ERROR at setup of test_ph_out_of_walltime ___________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_ph.py, line 47
  def test_ph_out_of_walltime(aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, data_regression):
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_ph.py:47
____________________ ERROR at setup of test_projwfc_default ____________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_projwfc.py, line 29
  def test_projwfc_default(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_projwfc.py:29
______________________ ERROR at setup of test_pw_default _______________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 34
  def test_pw_default(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:34
___________________ ERROR at setup of test_pw_default_no_xml ___________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 64
  def test_pw_default_no_xml(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:64
_________________ ERROR at setup of test_pw_default_xml_190304 _________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 100
  def test_pw_default_xml_190304(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:100
_________________ ERROR at setup of test_pw_default_xml_191206 _________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 130
  def test_pw_default_xml_191206(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:130
_______________ ERROR at setup of test_pw_initialization_xml_new _______________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 160
  def test_pw_initialization_xml_new(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:160
_________________ ERROR at setup of test_pw_failed_interrupted _________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 250
  def test_pw_failed_interrupted(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:250
_____________ ERROR at setup of test_pw_failed_interrupted_stdout ______________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 279
  def test_pw_failed_interrupted_stdout(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:279
_______________ ERROR at setup of test_pw_failed_interrupted_xml _______________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 311
  def test_pw_failed_interrupted_xml(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:311
_______________ ERROR at setup of test_pw_failed_out_of_walltime _______________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 360
  def test_pw_failed_out_of_walltime(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:360
_________ ERROR at setup of test_pw_failed_out_of_walltime_interrupted _________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 384
  def test_pw_failed_out_of_walltime_interrupted(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:384
______________ ERROR at setup of test_pw_failed_scf_not_converged ______________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 413
  def test_pw_failed_scf_not_converged(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:413
___________________ ERROR at setup of test_pw_relax_success ____________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 437
  def test_pw_relax_success(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:437
__________________ ERROR at setup of test_pw_vcrelax_success ___________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 511
  def test_pw_vcrelax_success(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:511
_____________ ERROR at setup of test_pw_vcrelax_success_fractional _____________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py, line 539
  def test_pw_vcrelax_success_fractional(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_inputs, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw.py:539
_____________________ ERROR at setup of test_pw2gw_default _____________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw2gw.py, line 9
  def test_pw2gw_default(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw2gw.py:9
_________________ ERROR at setup of test_pw2wannier90_default __________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw2wannier90.py, line 40
  def test_pw2wannier90_default(
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_pw2wannier90.py:40
______________________ ERROR at setup of test_q2r_default ______________________
file /home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_q2r.py, line 13
  def test_q2r_default(aiida_profile, fixture_localhost, generate_calc_job_node, generate_parser, data_regression):
E       fixture 'data_regression' not found
>       available fixtures: aiida_local_code_factory, aiida_localhost, aiida_profile, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, clear_database, clear_database_after_test, clear_database_before_test, cov, doctest_namespace, fixture_code, fixture_localhost, fixture_sandbox, generate_calc_job, generate_calc_job_node, generate_kpoints_mesh, generate_parser, generate_remote_data, generate_structure, generate_upf_data, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, temp_dir, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/parsers/test_q2r.py:13
=================================== FAILURES ===================================
_____________________________ test_create_builder ______________________________

aiida_profile = <aiida.manage.tests.TestManager object at 0x7faae22745b0>
fixture_sandbox = <aiida.common.folders.SandboxFolder object at 0x7faad9df2670>
fixture_code = <function fixture_code.<locals>._fixture_code at 0x7faadaa1fca0>
generate_upf_data = <function generate_upf_data.<locals>._generate_upf_data at 0x7faada0a3a60>
generate_calc_job = <function generate_calc_job.<locals>._generate_calc_job at 0x7faadaa1fc10>

    def test_create_builder(aiida_profile, fixture_sandbox, fixture_code, generate_upf_data, generate_calc_job):
        """Test the `create_builder_from_file` method that parses an existing `pw.x` folder into a process builder.
    
        The input file used is the one generated for `tests.calculations.test_pw.test_pw_default`.
        """
        entry_point_name = 'quantumespresso.pw'
        code = fixture_code(entry_point_name)
    
        metadata = {
            'options': {
                'resources': {
                    'num_machines': 1,
                    'num_mpiprocs_per_machine': 32,
                },
                'max_memory_kb': 1000,
                'max_wallclock_seconds': 60 * 60 * 12,
                'withmpi': True,
            }
        }
    
        in_foldername = os.path.join('tests', 'calculations', 'test_pw')
        in_folderpath = os.path.abspath(in_foldername)
    
        upf_foldername = os.path.join('tests', 'fixtures', 'pseudos')
        upf_folderpath = os.path.abspath(upf_foldername)
>       si_upf = generate_upf_data('Si')

tools/test_immigrate.py:34: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

element = 'Si'

    def _generate_upf_data(element):
        """Return `UpfData` node."""
        from aiida.orm import UpfData
    
        filename = os.path.join('tests', 'fixtures', 'pseudos', '{}.upf'.format(element))
        filepath = os.path.abspath(filename)
    
>       with io.open(filepath, 'r') as handle:
E       FileNotFoundError: [Errno 2] No such file or directory: '/home/werner/Public/hpc/tools/aiida/aiida-quantumespresso.git/tests/tests/fixtures/pseudos/Si.upf'

conftest.py:167: FileNotFoundError
=============================== warnings summary ===============================
/home/werner/.pyenv/versions/anaconda3-2020.02-py38/envs/aiida-py38/lib/python3.8/site-packages/past/builtins/misc.py:45
  /home/werner/.pyenv/versions/anaconda3-2020.02-py38/envs/aiida-py38/lib/python3.8/site-packages/past/builtins/misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
    from imp import reload

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== short test summary info ============================
FAILED tools/test_immigrate.py::test_create_builder - FileNotFoundError: [Err...
ERROR calculations/test_epw.py::test_epw_default
ERROR calculations/test_matdyn.py::test_matdyn_default
ERROR calculations/test_ph.py::test_ph_default
ERROR calculations/test_ph.py::test_ph_qpoint_list
ERROR calculations/test_pw.py::test_pw_default
ERROR calculations/test_pw2gw.py::test_pw_default
ERROR parsers/test_cp.py::test_cp_default
ERROR parsers/test_dos.py::test_dos_default
ERROR parsers/test_matdyn.py::test_matdyn_default
ERROR parsers/test_neb.py::test_neb_default
ERROR parsers/test_neb.py::test_neb_all_iterations
ERROR parsers/test_ph.py::test_ph_default
ERROR parsers/test_ph.py::test_ph_not_converged
ERROR parsers/test_ph.py::test_ph_out_of_walltime
ERROR parsers/test_projwfc.py::test_projwfc_default
ERROR parsers/test_pw.py::test_pw_default
ERROR parsers/test_pw.py::test_pw_default_no_xml
ERROR parsers/test_pw.py::test_pw_default_xml_190304
ERROR parsers/test_pw.py::test_pw_default_xml_191206
ERROR parsers/test_pw.py::test_pw_initialization_xml_new
ERROR parsers/test_pw.py::test_pw_failed_interrupted
ERROR parsers/test_pw.py::test_pw_failed_interrupted_stdout
ERROR parsers/test_pw.py::test_pw_failed_interrupted_xml
ERROR parsers/test_pw.py::test_pw_failed_out_of_walltime
ERROR parsers/test_pw.py::test_pw_failed_out_of_walltime_interrupted
ERROR parsers/test_pw.py::test_pw_failed_scf_not_converged
ERROR parsers/test_pw.py::test_pw_relax_success
ERROR parsers/test_pw.py::test_pw_vcrelax_success
ERROR parsers/test_pw.py::test_pw_vcrelax_success_fractional
ERROR parsers/test_pw2gw.py::test_pw2gw_default
ERROR parsers/test_pw2wannier90.py::test_pw2wannier90_default
ERROR parsers/test_q2r.py::test_q2r_default
============= 1 failed, 54 passed, 1 warning, 32 errors in 19.21s ==============
```

Any hints?
Regards

Hi, 

@sphuber I use a local git repo of this plugin to install it but not use the [dev] extra.  Now I tried with the following steps, but still failed.

From the top-level of this package's local git repo, do the following operations:

```
$ git pull
$ pyenv shell aiida-py38
$ pip install -e .[dev]
$ reentry scan 
```

There are too many errors, I just posted a little of the out here:
```

$ pytest
[...]
/home/werner/.pyenv/versions/anaconda3-2020.02-py38/envs/aiida-py38/lib/python3.8/site-packages/pgtest/pgtest.py:534: TimeoutError
______________________________ ERROR at setup of test_restart_ph ______________________________

    @pytest.fixture(scope='session', autouse=True)
    def aiida_profile():
        """Set up AiiDA test profile for the duration of the tests.
    
        Note: scope='session' limits this fixture to run once per session. Thanks to ``autouse=True``, you don't actually
         need to depend on it explicitly - it will activate as soon as you import it in your ``conftest.py``.
        """
>       with test_manager(backend=get_test_backend_name(), profile_name=get_test_profile_name()) as manager:

../aiida-core.git/aiida/manage/tests/pytest_fixtures.py:34: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/home/werner/.pyenv/versions/anaconda3-2020.02-py38/lib/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
../aiida-core.git/aiida/manage/tests/__init__.py:448: in test_manager
    _GLOBAL_TEST_MANAGER.use_temporary_profile(backend=backend, pgtest=pgtest)
../aiida-core.git/aiida/manage/tests/__init__.py:86: in use_temporary_profile
    mngr.create_profile()
../aiida-core.git/aiida/manage/tests/__init__.py:318: in create_profile
    self.create_aiida_db()
../aiida-core.git/aiida/manage/tests/__init__.py:299: in create_aiida_db
    self.create_db_cluster()
../aiida-core.git/aiida/manage/tests/__init__.py:289: in create_db_cluster
    self.pg_cluster = PGTest(**self._pgtest)
/home/werner/.pyenv/versions/anaconda3-2020.02-py38/envs/aiida-py38/lib/python3.8/site-packages/pgtest/pgtest.py:332: in __init__
    self._start_server()
/home/werner/.pyenv/versions/anaconda3-2020.02-py38/envs/aiida-py38/lib/python3.8/site-packages/pgtest/pgtest.py:429: in _start_server
    self._wait_for_server_ready(5)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = PGTest(database='postgres', username='postgres', port=48655, log_file='/tmp/tmpuk51rxef/pgtest_log.txt', no_cleanup=False, copy_cluster=None, cluster='/tmp/tmpuk51rxef', pg_ctl='/usr/lib/postgresql/11/bin/pg_ctl'), max_connections=11
wait = 5

    def _wait_for_server_ready(self, wait):
        """Sleep while we have no connection, timing out after `wait` seconds
    
            Args:
                wait - int, number of seconds to timeout the connection
            """
        endtime = datetime.datetime.utcnow() + datetime.timedelta(seconds=wait)
        while not self._is_connection_available():
            time.sleep(0.1)
            if datetime.datetime.utcnow() > endtime:
>               raise TimeoutError('Server failed to start')
E               pgtest.pgtest.TimeoutError: Server failed to start

/home/werner/.pyenv/versions/anaconda3-2020.02-py38/envs/aiida-py38/lib/python3.8/site-packages/pgtest/pgtest.py:534: TimeoutError
______________________________ ERROR at setup of test_restart_pw ______________________________

    @pytest.fixture(scope='session', autouse=True)
    def aiida_profile():
        """Set up AiiDA test profile for the duration of the tests.
    
        Note: scope='session' limits this fixture to run once per session. Thanks to ``autouse=True``, you don't actually
         need to depend on it explicitly - it will activate as soon as you import it in your ``conftest.py``.
        """
>       with test_manager(backend=get_test_backend_name(), profile_name=get_test_profile_name()) as manager:

../aiida-core.git/aiida/manage/tests/pytest_fixtures.py:34: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/home/werner/.pyenv/versions/anaconda3-2020.02-py38/lib/python3.8/contextlib.py:113: in __enter__
    return next(self.gen)
../aiida-core.git/aiida/manage/tests/__init__.py:448: in test_manager
    _GLOBAL_TEST_MANAGER.use_temporary_profile(backend=backend, pgtest=pgtest)
../aiida-core.git/aiida/manage/tests/__init__.py:86: in use_temporary_profile
    mngr.create_profile()
../aiida-core.git/aiida/manage/tests/__init__.py:318: in create_profile
    self.create_aiida_db()
../aiida-core.git/aiida/manage/tests/__init__.py:299: in create_aiida_db
    self.create_db_cluster()
../aiida-core.git/aiida/manage/tests/__init__.py:289: in create_db_cluster
    self.pg_cluster = PGTest(**self._pgtest)
/home/werner/.pyenv/versions/anaconda3-2020.02-py38/envs/aiida-py38/lib/python3.8/site-packages/pgtest/pgtest.py:332: in __init__
    self._start_server()
/home/werner/.pyenv/versions/anaconda3-2020.02-py38/envs/aiida-py38/lib/python3.8/site-packages/pgtest/pgtest.py:429: in _start_server
    self._wait_for_server_ready(5)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = PGTest(database='postgres', username='postgres', port=52361, log_file='/tmp/tmp8g7e5rv2/pgtest_log.txt', no_cleanup=False, copy_cluster=None, cluster='/tmp/tmp8g7e5rv2', pg_ctl='/usr/lib/postgresql/11/bin/pg_ctl'), max_connections=11
wait = 5

    def _wait_for_server_ready(self, wait):
        """Sleep while we have no connection, timing out after `wait` seconds
    
            Args:
                wait - int, number of seconds to timeout the connection
            """
        endtime = datetime.datetime.utcnow() + datetime.timedelta(seconds=wait)
        while not self._is_connection_available():
            time.sleep(0.1)
            if datetime.datetime.utcnow() > endtime:
>               raise TimeoutError('Server failed to start')
E               pgtest.pgtest.TimeoutError: Server failed to start

/home/werner/.pyenv/versions/anaconda3-2020.02-py38/envs/aiida-py38/lib/python3.8/site-packages/pgtest/pgtest.py:534: TimeoutError
================================= 87 error in 532.51 seconds =================================

```

Regards

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -12,6 +12,17 @@ import six
 pytest_plugins = ['aiida.manage.tests.pytest_fixtures']  # pylint: disable=invalid-name
 
 
+@pytest.fixture(scope='session')
+def filepath_tests():
+    """Return the absolute filepath of the `tests` folder.
+
+    .. warning:: if this file moves with respect to the `tests` folder, the implementation should change.
+
+    :return: absolute filepath of `tests` folder which is the basepath for all test resources.
+    """
+    return os.path.dirname(os.path.abspath(__file__))
+
+
 @pytest.fixture(scope='function')
 def fixture_sandbox():
     """Return a `SandboxFolder`."""
@@ -154,15 +165,14 @@ def generate_calc_job_node():
 
 
 @pytest.fixture(scope='session')
-def generate_upf_data():
+def generate_upf_data(filepath_tests):
     """Return a `UpfData` instance for the given element a file for which should exist in `tests/fixtures/pseudos`."""
 
     def _generate_upf_data(element):
         """Return `UpfData` node."""
         from aiida.orm import UpfData
 
-        filename = os.path.join('tests', 'fixtures', 'pseudos', '{}.upf'.format(element))
-        filepath = os.path.abspath(filename)
+        filepath = os.path.join(filepath_tests, 'fixtures', 'pseudos', '{}.upf'.format(element))
 
         with io.open(filepath, 'r') as handle:
             upf = UpfData(file=handle.name)

--- a/tests/tools/test_immigrate.py
+++ b/tests/tools/test_immigrate.py
@@ -6,7 +6,9 @@ import os
 from aiida_quantumespresso.tools.pwinputparser import create_builder_from_file
 
 
-def test_create_builder(aiida_profile, fixture_sandbox, fixture_code, generate_upf_data, generate_calc_job):
+def test_create_builder(
+    aiida_profile, fixture_sandbox, fixture_code, generate_upf_data, generate_calc_job, filepath_tests
+):
     """Test the `create_builder_from_file` method that parses an existing `pw.x` folder into a process builder.
 
     The input file used is the one generated for `tests.calculations.test_pw.test_pw_default`.
@@ -26,11 +28,9 @@ def test_create_builder(aiida_profile, fixture_sandbox, fixture_code, generate_u
         }
     }
 
-    in_foldername = os.path.join('tests', 'calculations', 'test_pw')
-    in_folderpath = os.path.abspath(in_foldername)
+    in_folderpath = os.path.join(filepath_tests, 'calculations', 'test_pw')
+    upf_folderpath = os.path.join(filepath_tests, 'fixtures', 'pseudos')
 
-    upf_foldername = os.path.join('tests', 'fixtures', 'pseudos')
-    upf_folderpath = os.path.abspath(upf_foldername)
     si_upf = generate_upf_data('Si')
     si_upf.store()
 

